Foundation/NSJSONSerialization.swift

```diff
@@ -281,6 +291,21 @@ internal extension JSONSerialization {
     }
 }
 
+private protocol AnyOptional {
+    var value: Any? { get }
+}
+
+extension Optional : AnyOptional {
+    public var value: Any? {
+        switch self {
+        case .none:
+            return nil
+        case .some(let x):
+            return x
+        }
+    }
+}
+
```

Comment:
the problem is that

    1> let maybeString: String? = nil
     maybeString: String? = nil
     2> let maybeStringAsAny: Any = maybeString
     maybeStringAsAny: Any = {
      payload_data_0 = 0x0000000000414de0
      payload_data_1 = 0x0000000000000000
      payload_data_2 = 0x0000000000000000
      instance_type = 0x0000000000414ce0
    }
    3> switch maybeStringAsAny as Any? { case .none: print("none"); default: print("some") }
     some

so the protocol `AnyOptional` is basically just to check whether the `Any` value is some form of optional. If you go from `String?` to `Any` (which is done through the function call) and then from `Any` to `Any?` it will _always_ be `.some`:

      7> Optional<String>.none as Any as Any? 
    $R2: Any? = some {
      payload_data_0 = 0x0000000000416420
      payload_data_1 = 0x0000000000000000
      payload_data_2 = 0x00007fffffffed20 -> 0x0000000000400960 repl_swift`__libc_csu_init
      instance_type = 0x0000000000414ce0
    }